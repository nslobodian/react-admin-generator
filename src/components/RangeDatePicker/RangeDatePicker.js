import React from 'react'
import 'bootstrap-daterangepicker/daterangepicker.css'
import DatetimeRangePicker from 'react-bootstrap-datetimerangepicker'
import FaClockO from 'react-icons/lib/fa/clock-o'
import FaClose from 'react-icons/lib/fa/close'
import { convertDate } from 'utils/helpers'
import s from './RangeDatePicker.scss'

export default ({ input: { onChange, value: { startDate, endDate }, name }, meta: { touched, invalid }, timePicker = true, disabled, label, singleDatePicker = false, placeholder = 'Any time', customTrigger, customOnChange }) =>
  <div className={disabled && s.disabled}>
    {label && <div className='text-muted'>
      {label}
    </div>}
    <div className={s.DatetimeRangePicker}>
      {(!customTrigger && !label) && <div className={s.SVGWrapper}>
        {startDate && <FaClose className={s.FaClose} onClick={() => {
          onChange({})
        }} /> }
        <FaClockO className={startDate && s.FaClockO} />
      </div>}
      <DatetimeRangePicker
        timePicker={timePicker}
        timePicker24Hour
        showDropdowns
        timePickerSeconds
        startDate={startDate}
        endDate={singleDatePicker ? startDate : endDate}
        singleDatePicker={singleDatePicker}
        onApply={(j, { endDate, startDate }) => (customOnChange || onChange)({ endDate, startDate })}
      >
        {label ? (
          <div className='input'>
            <div className='form-control'>{startDate && convertDate(startDate, true)}</div>
          </div>
          )
          : (customTrigger || startDate
          ? <div className={s.ButtonWrapper}>
            <div>{startDate.format('DD.MM.YYYY HH:mm')}</div>
            {!singleDatePicker && <div>{convertDate(endDate, false)}</div>}
          </div>
          : <div className={s.ButtonWrapper}>
            <div>{placeholder}</div>
          </div>)
        }
      </DatetimeRangePicker>
      {touched && invalid &&
      <div className='alert alert-danger m-t-1' role='alert'>{name} is required</div>}
    </div>
  </div>
