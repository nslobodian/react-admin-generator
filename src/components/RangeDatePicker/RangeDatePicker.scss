.DatetimeRangePicker {
  cursor: pointer;
  position: relative;
  border-width: 2px;
  border-color: black;


  .SVGWrapper {
    position: absolute;
    top: 0;

    .FaClose {
      display: none;
      z-index: 2;
    }

    &:hover {
      .FaClose {
        display: inline;
      }

      .FaClockO {
        display: none;
      }
    }
  }

  .ButtonWrapper {
    display: flex;
    flex-direction: column;
    margin-left: 20px;
    color: #777;
  }
}

.disabled {
  cursor: not-allowed !important;
  pointer-events: none !important;
}