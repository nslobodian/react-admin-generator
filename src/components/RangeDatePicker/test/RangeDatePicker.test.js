import React from 'react'
import { mount } from 'enzyme'
import toJson from 'enzyme-to-json'
import RangeDatePicker from '../RangeDatePicker'

describe('(Components) RangeDatePicker', () => {
  it('Should render component properly', () => {

    const props = {
      input: {
        onChange: jest.fn(),
        value: {
          startDate: {
            format: jest.fn(),
          },
          endDate: {
            format: jest.fn(),
          },
        },
        name: 'Name',
      },
      meta: {
        touched: true,
        invalid: true,
      },
      singleDatePicker: false,
      placeholder: 'Any time',
      customTrigger: false,
      customOnChange: jest.fn(),
    }
    const wrapper = mount(
      <RangeDatePicker
        {...props}
      />)
    expect(toJson(wrapper)).toMatchSnapshot()
    wrapper.find('FaClose').simulate('click')
  }),
  it('Should render component properly without singleDatePicker & placeholder & startDate', () => {

    const props = {
      input: {
        onChange: jest.fn(),
        value: {
          endDate: {
            format: jest.fn(),
          },
        },
        name: 'Name',
      },
      meta: {
        touched: true,
        invalid: true,
      },
      customTrigger: false,
      customOnChange: jest.fn(),
    }
    const wrapper = mount(
      <RangeDatePicker
        {...props}
      />)
    expect(toJson(wrapper)).toMatchSnapshot()
  })
})
