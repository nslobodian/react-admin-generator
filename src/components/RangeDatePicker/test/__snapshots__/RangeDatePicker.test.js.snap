// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`(Components) RangeDatePicker Should render component properly 1`] = `
<Component
  customOnChange={[Function]}
  customTrigger={false}
  input={
    Object {
      "name": "Name",
      "onChange": [Function],
      "value": Object {
        "endDate": Object {
          "format": [Function],
        },
        "startDate": Object {
          "format": [Function],
        },
      },
    }
  }
  meta={
    Object {
      "invalid": true,
      "touched": true,
    }
  }
  placeholder="Any time"
  singleDatePicker={false}
>
  <div
    className="DatetimeRangePicker"
  >
    <div
      className="SVGWrapper"
    >
      <FaClose
        className="FaClose"
        onClick={[Function]}
      >
        <IconBase
          className="FaClose"
          onClick={[Function]}
          viewBox="0 0 40 40"
        >
          <svg
            className="FaClose"
            fill="currentColor"
            height="1em"
            onClick={[Function]}
            preserveAspectRatio="xMidYMid meet"
            style={
              Object {
                "color": undefined,
                "verticalAlign": "middle",
              }
            }
            viewBox="0 0 40 40"
            width="1em"
          >
            <g>
              <path
                d="m33.5 29.5q0 0.9-0.7 1.5l-3 3.1q-0.6 0.6-1.5 0.6t-1.5-0.6l-6.6-6.6-6.5 6.6q-0.7 0.6-1.6 0.6t-1.5-0.6l-3-3.1q-0.6-0.6-0.6-1.5t0.6-1.5l6.5-6.6-6.5-6.5q-0.6-0.7-0.6-1.6t0.6-1.5l3-3q0.6-0.6 1.5-0.6t1.6 0.6l6.5 6.6 6.6-6.6q0.6-0.6 1.5-0.6t1.5 0.6l3.1 3q0.6 0.7 0.6 1.5t-0.6 1.6l-6.6 6.5 6.6 6.6q0.6 0.6 0.6 1.5z"
              />
            </g>
          </svg>
        </IconBase>
      </FaClose>
      <FaClockO
        className="FaClockO"
      >
        <IconBase
          className="FaClockO"
          viewBox="0 0 40 40"
        >
          <svg
            className="FaClockO"
            fill="currentColor"
            height="1em"
            preserveAspectRatio="xMidYMid meet"
            style={
              Object {
                "color": undefined,
                "verticalAlign": "middle",
              }
            }
            viewBox="0 0 40 40"
            width="1em"
          >
            <g>
              <path
                d="m23 12.1v10q0 0.4-0.2 0.6t-0.5 0.2h-7.2q-0.3 0-0.5-0.2t-0.2-0.6v-1.4q0-0.3 0.2-0.5t0.5-0.2h5v-7.9q0-0.3 0.2-0.5t0.6-0.2h1.4q0.3 0 0.5 0.2t0.2 0.5z m9.3 7.9q0-3.3-1.6-6.1t-4.5-4.4-6.1-1.6-6.1 1.6-4.4 4.4-1.6 6.1 1.6 6.1 4.4 4.4 6.1 1.6 6.1-1.6 4.5-4.4 1.6-6.1z m5 0q0 4.7-2.3 8.6t-6.3 6.2-8.6 2.3-8.6-2.3-6.2-6.2-2.3-8.6 2.3-8.6 6.2-6.2 8.6-2.3 8.6 2.3 6.3 6.2 2.3 8.6z"
              />
            </g>
          </svg>
        </IconBase>
      </FaClockO>
    </div>
    <DatetimeRangePicker
      endDate={"2017-09-13T22:00:00.000Z"}
      onApply={[Function]}
      showDropdowns={true}
      singleDatePicker={false}
      startDate={"2017-09-13T22:00:00.000Z"}
      timePicker={true}
      timePicker24Hour={true}
      timePickerSeconds={true}
    >
      <div>
        <div
          className="ButtonWrapper"
        >
          <div />
          <div />
        </div>
      </div>
    </DatetimeRangePicker>
    <div
      className="alert alert-danger m-t-1"
      role="alert"
    >
      Name
       is required
    </div>
  </div>
</Component>
`;

exports[`(Components) RangeDatePicker Should render component properly without singleDatePicker & placeholder & startDate 1`] = `
<Component
  customOnChange={[Function]}
  customTrigger={false}
  input={
    Object {
      "name": "Name",
      "onChange": [Function],
      "value": Object {
        "endDate": Object {
          "format": [Function],
        },
      },
    }
  }
  meta={
    Object {
      "invalid": true,
      "touched": true,
    }
  }
>
  <div
    className="DatetimeRangePicker"
  >
    <div
      className="SVGWrapper"
    >
      <FaClockO>
        <IconBase
          viewBox="0 0 40 40"
        >
          <svg
            fill="currentColor"
            height="1em"
            preserveAspectRatio="xMidYMid meet"
            style={
              Object {
                "color": undefined,
                "verticalAlign": "middle",
              }
            }
            viewBox="0 0 40 40"
            width="1em"
          >
            <g>
              <path
                d="m23 12.1v10q0 0.4-0.2 0.6t-0.5 0.2h-7.2q-0.3 0-0.5-0.2t-0.2-0.6v-1.4q0-0.3 0.2-0.5t0.5-0.2h5v-7.9q0-0.3 0.2-0.5t0.6-0.2h1.4q0.3 0 0.5 0.2t0.2 0.5z m9.3 7.9q0-3.3-1.6-6.1t-4.5-4.4-6.1-1.6-6.1 1.6-4.4 4.4-1.6 6.1 1.6 6.1 4.4 4.4 6.1 1.6 6.1-1.6 4.5-4.4 1.6-6.1z m5 0q0 4.7-2.3 8.6t-6.3 6.2-8.6 2.3-8.6-2.3-6.2-6.2-2.3-8.6 2.3-8.6 6.2-6.2 8.6-2.3 8.6 2.3 6.3 6.2 2.3 8.6z"
              />
            </g>
          </svg>
        </IconBase>
      </FaClockO>
    </div>
    <DatetimeRangePicker
      endDate={"2017-09-13T22:00:00.000Z"}
      onApply={[Function]}
      showDropdowns={true}
      singleDatePicker={false}
      startDate={"2017-09-14T08:53:31.686Z"}
      timePicker={true}
      timePicker24Hour={true}
      timePickerSeconds={true}
    >
      <div>
        <div
          className="ButtonWrapper"
        >
          <div>
            Any time
          </div>
        </div>
      </div>
    </DatetimeRangePicker>
    <div
      className="alert alert-danger m-t-1"
      role="alert"
    >
      Name
       is required
    </div>
  </div>
</Component>
`;
